### ISSUE TITLE
Feature selection for binary classification

### ISSUE THREAD

**Issue by haboryme** [Microsoft/LightGBM#746]
I'm not really sure if this is an issue or a lack of understanding on my part, my apologies if it is.

LightGBM is used for binary classification and for identifying the relevant features.
I'm using a dataset that has 2 features (index : 1 and 7) that are equal (100% correlation) and perfectly predict the labels whereas the others are randomly generated.

- If feature_fraction = 1 the feature coming first in the index (1) is always used to perfectly predict the labels. Feature 7 never shows up as relevant.

- If feature_fraction < 1 then the second relevant feature also appear but always with a lesser importance than the first one.

I tried running the classification many times and the code below shows that overall feature_1 has a greater importance than feature_7 even though they are equal.
Shouldn't the importance average out to similar values over the number of runs? Or am I missing something obvious?

## Environment info
Operating System: Windows 10 - 64bit
CPU: i7-2670QM
Python version: 3.5.2

## Python 3.5 code
```
import lightgbm as lgb
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
import matplotlib.pyplot as plt
df = pd.DataFrame(np.random.randn(1000).reshape(10,100), columns=[np.repeat(["a","b"],50)])
df.iloc[[1,7],df.columns.get_loc("a")] = 0
df.iloc[[1,7],df.columns.get_loc("b")] = 1
df.loc["state"] = np.repeat([-1,1],50)       
train, test = train_test_split(range(df.shape[1]))
df_train = df.iloc[:,train]
df_test = df.iloc[:,test]
y_train = df_train.loc["state"].values
y_test = df_test.loc["state"].values
X_train = df_train.drop("state", axis=0).transpose()
X_test = df_test.drop("state", axis=0).transpose()
num_train, num_feature = X_train.shape
lgb_train = lgb.Dataset(X_train, y_train, free_raw_data=False)
lgb_eval = lgb.Dataset(X_test, y_test, reference=lgb_train,free_raw_data=False)
overall_importance = []
for i in range(300):
    params = {
    'boosting_type': 'gbdt',
    'objective': 'binary',
    'metric': 'binary_logloss',
    'num_leaves': 3,
    'learning_rate': 0.01,
    'feature_fraction': 0.7,
    'bagging_fraction': 0.8,
    "feature_fraction_seed":i,
    'bagging_freq': 1,
    'verbose': 0
}
    gbm = lgb.train(params,
                    lgb_train,
                    num_boost_round=10,
                    valid_sets=lgb_train,  # eval training data
                    feature_name=feature_name,
                    categorical_feature=[10])
    overall_importance.append(gbm.feature_importance())

overall_importance = pd.DataFrame(overall_importance)
feature_name = ['feature_' + str(col) for col in range(num_feature)]
means = overall_importance.apply(np.mean, axis=0)
stds = overall_importance.apply(np.std, axis=0)
fig, ax = plt.subplots()
ax.barh(range(10), means, xerr=stds, align='center', ecolor='black')
ax.set_yticks(range(10))
ax.set_yticklabels(feature_name)
ax.invert_yaxis()  
ax.set_xlabel('Importance')
plt.show()
```

**Comment by guolinke** [comment]
@haboryme 
this is by design. when two features are the same, LightGBM will always use the one with smaller index.
When using feature_fraction, the two features could be not used at the same time, so it is possible to use the one with large index.
